test/functionTest.php

```diff
@@ -4,15 +4,20 @@
 
 class functionTest extends TestCase {
 
-    public function testSuccessloopCartoons() {
+    public function testSuccessdisplayCartoons() {
         $expected = "<div class='eachCharacter'><h2 class='charName'>Test Name</h2><p class='cardContent'>From: Test Show</p><p class='cardContent'>Species: Test Species</p><p class='cardContent'>Catchphrase: Test Phrase</p><img src=Test Image></div>";
-        $actual = loopCartoons([["name" => "Test Name", "species" => "Test Species", "tvshow" => "Test Show", "catchphase" => "Test Phrase", "image" => "Test Image"]]);
+        $actual = displayCartoons([["name" => "Test Name", "species" => "Test Species", "tvshow" => "Test Show", "catchphase" => "Test Phrase", "image" => "Test Image"]]);
 
         $this->assertEquals($expected, $actual);
     }
 
     public function testErrorgivenwrongInput(){
         $this->expectException(TypeError::class);
-        loopCartoons('test');
+        displayCartoons('test');
+    }
+
+    public function testErrorwrongArrKey(){
+        $this->expectException(TypeError::class);
+        displayCartoons(6);
```

Comment:
sorry, not what I meant. what happens if you pass it an array like this:
`[["notname" => "Test Name", "notspecies" => "Test Species", "nottvshow" => "Test Show", "notcatchphase" => "Test Phrase", "notimage" => "Test Image"]]`

the data structure is correct but the keys are wrong (contain the word not), what will your function do then?